routes/api.js

```diff
@@ -431,7 +433,13 @@ router.get('/create_account', async (req, res) => {
         const user = await req.db.users.findOne({ where: { email: decoded.email } });
         if (!user) {
           res.status(400).json({ error: 'error_api_user_exists_not' });
+        } else if (user.creation_hash !== null) {
+          res.status(400).json({ error: 'error_api_account_creation_progress' });
         } else if (user.status === 'approved') {
+          const creationHash = hash.sha256(crypto.randomBytes(32)).toString('hex');
+          await req.db.users.update({
+            creation_hash: creationHash,
+          }, { where: { email: decoded.email } });
```

Comment:
Can you add condition inside `where` to update only user with empty `create_hash`? This way we are sure if there is 2 requests in same time one of would fail to store an hash.